src/cluster_legacy.c

```diff
@@ -7131,7 +7136,7 @@ int clusterCommandSpecial(client *c) {
         /* CLUSTER FORGET <NODE ID> */
         clusterNode *n = clusterLookupNode(c->argv[2]->ptr, sdslen(c->argv[2]->ptr));
         if (!n) {
-            if (clusterBlacklistExists((char *)c->argv[2]->ptr))
+            if (clusterBlacklistExists((char *)c->argv[2]->ptr, sdslen(c->argv[2]->ptr)))
```

Comment:
I guess Ping wants to return here, something like this (i also like this)
```suggestion
            if (sdslen(sdslen(c->argv[2]->ptr)) == CLUSTER_NAMELEN && clusterBlacklistExists((char *)c->argv[2]->ptr))
```

Reply:
I am fine with that as well, although making the check inside clusterBlacklistExists is more defensive. However I have no special argument against any of the suggestions.

Reply:
we can also just add the len argument to the clusterBlacklistExists and have it miss finding non existing nodes in the blacklist. this would make the code much more correct, but would keep the unneeded sds allocation and dictionary search as they are today.

Reply:
@enjoy-binbin I change as per my last comment. If you still think we should add another check for cluster forget input, I am fine with it (just to prevent the not-so-interesting case of user who provides a very long node-id and bursting memory usage for the sds copy).